src/datasource.ts

```diff
@@ -315,6 +323,26 @@ export default class CogniteDatasource {
         };
         if (target.tab === Tab.Custom && target.assetQuery.func) {
           query.function = target.assetQuery.func.replace(/ID/g, String(ts.id));
+          const regexSearch = /\[.*?\]/g;
+          const regexMatches = query.function.match(regexSearch);
+          query.aliases = [];
+          for (const match of regexMatches) {
+            // format is id, aggregation, granularity
+            const aliasParts = match
+              .substr(1, match.length - 2)
+              .split(',')
```

Comment:
Keep in mind that `','.split(',')` yields `['', '']`. Might want to do an `.filter(string => string.length)` or similar.

Reply:
fixed